pkg/proxy/endpointslicecache_test.go

```diff
@@ -370,7 +370,7 @@ func TestEsDataChanged(t *testing.T) {
 				ObjectMeta: objMeta,
 				Ports:      []discovery.EndpointPort{port80, port443},
 			},
-			expectChanged: false,
+			expectChanged: true,
```

Comment:
See the commit message (and the discussion between Antonio and me above). The code is currently trying to optimize for something that will never happen in most clusters. The assumption is that removing this code will make us do _less_ work, because it was previously doing lots of work to try to avoid a case that basically never happened.